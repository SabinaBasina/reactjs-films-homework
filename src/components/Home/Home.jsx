import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './Home.scss';
import TvShowPage from '../TvShowPage';
import Loading from './Loading.gif';

class Home extends Component {
  increment = () => {
    const { onIncrement } = this.props;
    onIncrement();
  }

  decrement = () => {
    const { onDecrement } = this.props;
    onDecrement();
  }

  render() {
    const { page, tvShows } = this.props;
    return (

      <main>

        <TvShowPage page={page} />

        {!tvShows
          && (
            <img
              src={Loading}
              className={styles.Ready}
              alt="Loading page"
            />
          )}

        {tvShows
          && (
            <div className={styles.Pager}>

              {page > 0
                && (
                  <button type="button" onClick={e => this.decrement(e)}>
                    &larr; Previous Page
                  </button>
                )
              }

              <button type="button" onClick={e => this.increment(e)}>
                Next Page &rarr;
              </button>

            </div>
          )}


      </main>

    );
  }
}

export default Home;

Home.propTypes = {
  page: PropTypes.number,
  tvShows: PropTypes.arrayOf(PropTypes.array).isRequired,
  onIncrement: PropTypes.func,
  onDecrement: PropTypes.func,
};

Home.defaultProps = {
  page: 0,
  onIncrement: () => { },
  onDecrement: () => { },
};
