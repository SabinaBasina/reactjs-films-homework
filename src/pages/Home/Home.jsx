/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import styles from './Home.scss';
import TvShowPage from '../../components/TvShowPage';
import Loader from '../../components/Loader';

class Home extends Component {
  render() {
    const { isReadyTvShows, pageNumber } = this.props;

    return (

      <main>

        <TvShowPage page={pageNumber} />

        <Loader loading={!isReadyTvShows} />

        {isReadyTvShows
          && (
            <div className={styles.pager}>

              {pageNumber > 0
                && (
                  <Link to={`/page/${pageNumber - 1}`}>
                    <button type="button">
                      &larr; Previous Page
                    </button>
                  </Link>
                )
              }

              <Link to={`/page/${pageNumber + 1}`}>
                <button type="button">
                  Next Page &rarr;
                </button>
              </Link>

            </div>
          )
        }


      </main>

    );
  }
}

export default Home;

Home.propTypes = {
  isReadyTvShows: PropTypes.bool,
  pageNumber: PropTypes.number,
};

Home.defaultProps = {
  isReadyTvShows: false,
  pageNumber: 0,
};
