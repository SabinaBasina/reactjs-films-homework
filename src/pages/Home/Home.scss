@import '../TvShowDetails/TvShowDetails.scss';

.Pager {
  text-align: center;
  margin-top: 50px;
  margin-bottom: 80px;
}

.Pager button {
  outline: none;
  border-radius: 25px;
  width: 140px;
  height: 45px;
  text-align: center;
  color: #9d9d9d;
  background-color: #222;
  border: none;
  font-weight: 500;
  font-size: 15px;
  margin: 0 13% 15px 13%;
}

.Pager button:hover {
  color: white;
}
