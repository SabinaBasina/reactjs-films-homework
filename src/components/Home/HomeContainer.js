import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import Home from './Home';

const mapStateToProps = createStructuredSelector({
  page: state => state.page.page,
  tvShows: state => state.page.tvShows,
});

const mapDispatchToProps = {
  onIncrement: () => ({ type: 'INCREMENT' }),
  onDecrement: () => ({ type: 'DECREMENT' }),
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
