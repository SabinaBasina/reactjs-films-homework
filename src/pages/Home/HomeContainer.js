import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import Home from './Home';
import { getIsReadyTvShows } from '../../modules/moduleHome/moduleHomeSelectors';

const mapStateToProps = createStructuredSelector({
  isReadyTvShows: getIsReadyTvShows,
});

export default connect(mapStateToProps)(Home);
