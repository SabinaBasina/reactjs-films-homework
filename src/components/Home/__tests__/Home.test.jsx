import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Home from '../Home';

/* global test, expect */

test('Home test', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Home />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
