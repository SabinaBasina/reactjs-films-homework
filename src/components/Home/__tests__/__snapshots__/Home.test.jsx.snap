// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test 1`] = `
<main>
  <Connect(TvShowPage)
    page={0}
  />
  <div
    className="Pager"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Next Page â†’
    </button>
  </div>
</main>
`;
